### DISCUSSION TITLE
Neopixel/WS2812 effects?

### DISCUSSION THREAD

**Discussion by strofos** [arendst/Tasmota#10833] [Q&A] [answered]
Sorry if this can be an off topic or stupid question! I am a noob on Tasmota. 3 days ago I just flashed my first device and it worked perfectly. 2 days ago I completed my first installation of Home Assistent on a RPi3 and integrated the Tasmota dev (an aquarium IoT based on Generic GPIO, Relay, Switch, RGB/Neopixel) on it. 

I searched for effects like fade in, fade out, rainbow on the Neopixel integrations and I can't find something (directly in Tasmota, not NodeRed). How can I add an EffectStatus = None|FadeIn|FadeOut|Rainbow (like these https://github.com/kitesurfer1404/WS2812FX ) inside the xlgt_01_ws2812.ino (I think this is the one that gets the topic from a MQTT command) and make it available on the IDE? Sorry that I still don't understand the Tasmota engine! I'm new to it, but I am used to standard Arduino  and HAL like software/libraries.

**Comment by sfromis** [comment, marked as answer]
You do already have some built-in lighting controls in Tasmota, allowing you to control things like color, brightness, fading and a few effects schemes. But if you want extensive neopixel effects, WLED may be a better project for you.
https://tasmota.github.io/docs/Commands/#light

**Reply by strofos** [reply]
TKX @sfromis, read the command wiki and it was great! Do you know a way to easily integrate these commands in Home Assistant? I already use the MQTT broker. So for my understanding, I can do an /cmnd/aquarium/scheme%207 push to trigger the rainbow coloring. But how can I add a dropdown or a button to the for thisin the UI? 

Also maybe my Tasmota version (I build the dev on the GitHub) does not come with all those schemes in place. This is my .ino header. Also the scheme array reflects this:

/*********************************************************************************************\
 * WS2812 RGB / RGBW Leds using NeopixelBus library
 *
 * light_scheme  WS2812  3+ Colors  1+2 Colors  Effect
 * ------------  ------  ---------  ----------  -----------------
 *  0            yes     no         no          Clock
 *  1            yes     no         no          Incandescent
 *  2            yes     no         no          RGB
 *  3            yes     no         no          Christmas
 *  4            yes     no         no          Hanukkah
 *  5            yes     no         no          Kwanzaa
 *  6            yes     no         no          Rainbow
 *  7            yes     no         no          Fire
 *
\*********************************************************************************************/

**Reply by barbudor** [reply]
@strofos You wrote:
> I can do an /cmnd/aquarium/scheme%207 push

Beware that in MQTT there is 2 parts in the message:
- the topic, here `cmnd/aquarium/scheme`
- and the payload `7`

It is unlikely that your home automation tool requires both to be together in the same string with a URL-encoded space
May be a standard space separator

For any question regarding Home Assistant that are not directly related to Tasmota integration (such as how to make that in the HA GUI), you would find better support in one of the Home Assistant dedicated forums

**Reply by sfromis** [reply]
The source you can download from Github does contain the features from the docs, as long as you perform the commands the right way. I'd suggest first to play around in the console, to "get" how the commands work. The precompiled builds also have the neopixel features, except that some variations of neopixel may still need you to change settings and compile.

**Reply by strofos** [reply]
Thanks, I have to compile it as I do have LEDs with a different color order than the default. I have RGB instead of the default GRB. I also want to integrate a fish feeder based on the A4988 shield and an old CD stepper. I've found the Rotation method and I wanted to create an alias Feed = Rotate 90. Thus I am playing first with RGBs to better understand the Tasmota library.
